### PR TITLE
update the docs of how to run e2e tests

### LINKED ISSUE #846: Update docs of E2E tests in knative-gcp
**Problem**
 The docs of E2E tests in knative-gcp is out-dated and kind of confusing and it also misses some context info. 
For example: Figure the correct version of eventing to use in e2e tests and write docs. Latest nightly?  Latest prior to the vendored version of eventing?  First version after vendored version(or latest if none exists)? 

**[Persona:](https://github.com/knative/eventing/blob/master/docs/personas.md)** 
Which persona is this feature for?

**Exit Criteria**
A measurable (binary) test that would indicate that the problem has been resolved.

**Time Estimate (optional):** 
How many developer-days do you think this may take to resolve?

**Additional context (optional)**
Add any other context about the feature request here.

### PR DESCRIPTION
**Pull request by capri-xiyue**
Fixes #846 

## Proposed Changes

- update the docs of how to run e2e tests with existing cluster
- update the docs of how to run e2e tests with new cluster
-

**Release Note**

```release-note
Will add the docs of run e2e tests with newly created cluster later
```

**Docs**

### CODE REVIEW

File: test/e2e/README.md [hunk 1]
```diff
@@ -24,11 +24,25 @@ knative-gcp should be added under [knative-gcp e2e test lib](lib).
...
-running environment that meets
-[the e2e test environment requirements](#environment-requirements), and you need
-to specify the build tag `e2e`.
+### Prerequisites
 
+1. A running Kubernetes cluster with [knative-gcp](../../docs/install/install-knative-gcp.md) installed and configured.
+2. [Pub/Sub Enabled Service Account](../../docs/install/pubsub-service-account.md) installed.
+3. [Broker with Pub/Sub Channel](../../docs/install/install-broker-with-pubsub-channel.md) installed.
+4. A docker repo containing [the test images](#test-images). Remember to specify the build tag `e2e`.
+5. (Optional) Note that if you plan on running metrics-related E2E tests using the StackDriver
```
Comment (capri-xiyue) [inline test/e2e/README.md:33]: Anyone tried to run E2E tests with `TestChannelTracing`.
When I tried to run `E2E_PROJECT_ID=$PROJECT_ID  go test --tags=e2e ./test/e2e/... -run TestChannelTracing`, it failed with `Error retrieving Zipkin pod details: no zipkin Pod found on the cluster. Ensure monitoring is switched on for your Knative Setup zipkin.go:35: Unable to set up Zipkin for tracking`.
I already give the service account the `Monitoring Editor` role. Do I need to do other set up besides the Monitoring Editor role? In addition, the service account should be the service account of Data Plane, right?

Comment (capri-xiyue) [reply test/e2e/README.md]: @ian-mi  Do you know how to set up this Zipkin?

File: test/e2e/README.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
 # E2E Tests
 
-Prow will run `./e2e-tests.sh`.
+Prow will run `./e2e-tests.sh` with authentication mechanism using Kubernetes Secrets and `./e2e-wi-tests.sh` with authentication mechanism using Workload Identity.
```
Comment (chizhg) [inline test/e2e/README.md:3]: It's better to point out that Workload Identity is GKE specific and add the link https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity here

File: test/e2e/README.md [hunk 3]
```diff
@@ -24,11 +24,36 @@ knative-gcp should be added under [knative-gcp e2e test lib](lib).
 
 ## Running E2E Tests on an existing cluster
 
-To run [the e2e tests](../e2e) with `go test` command, you need to have a
-running environment that meets
-[the e2e test environment requirements](#environment-requirements), and you need
-to specify the build tag `e2e`.
+### Prerequisites
```
Comment (chizhg) [inline test/e2e/README.md:27]: It seems Matt didn't enable the markdown checker for this repo, and clearly the markdown file is not formatted.
Could you please manually run `prettier --write --prose-wrap=always $(find -name '*.md' | grep -v vendor | grep -v .github | grep -v docs/cmd/)` to format this PR?

File: test/e2e/README.md [hunk 4]
```diff
@@ -24,11 +24,36 @@ knative-gcp should be added under [knative-gcp e2e test lib](lib).
...
+There are two ways to set up authentication mechanism. 
+- If you want to run E2E tests with authentication mechanism using **Workload Identity**, please configure the authentication mechanism with **Workload Identity**.
```
Comment (chizhg) [inline test/e2e/README.md:29]:
```suggestion
- If you want to run E2E tests with authentication mechanism using **Workload Identity**, please follow below instructions to configure the authentication mechanism with **Workload Identity**.
```

File: test/e2e/README.md [hunk 5]
```diff
@@ -24,11 +24,36 @@ knative-gcp should be added under [knative-gcp e2e test lib](lib).
...
+- If you want to run E2E tests with authentication mechanism using **Kubernetes Secrets**, please configure the authentication mechanism with **Kubernetes Secrets**.
```
Comment (chizhg) [inline test/e2e/README.md:30]:
```suggestion
- If you want to run E2E tests with authentication mechanism using **Kubernetes Secrets**, please follow below instructions to configure the authentication mechanism with **Kubernetes Secrets**.
```

File: test/e2e/README.md [hunk 6]
```diff
@@ -104,6 +165,13 @@ For images deployed in GCR, a docker tag is mandatory to avoid issues with using
 ./test/upload-test-images.sh ./vendor/knative.dev/eventing/test/test_images/ e2e
 ```
 
+To run the script for all end to end test images:
+
+```bash
+./test/upload-test-images.sh ./test/test_images
+./test/upload-test-images.sh ./vendor/knative.dev/eventing/test/test_images/
```
Comment (chizhg) [inline test/e2e/README.md:172]: https://github.com/google/knative-gcp/blob/443a9e865c8e0e1169aaa5da766a0898c555dfa2/test/e2e-common.sh#L63 needs to be run before upload eventing test images

File: test/e2e/README.md [hunk 7]
```diff
@@ -42,35 +67,78 @@ go test --tags=e2e ./test/e2e/... --count=3
...
-1. A running Kubernetes cluster with [knative-gcp](../../docs/install) installed
-   and configured
-1. A docker repo containing [the test images](#test-images)
+If you want to run E2E tests with authentication mechanism using **Kubernetes Secrets**:
+```shell
+./test/e2e-tests.sh
+```
+If you want to run E2E tests with authentication mechanism using **Workload Identity**:
+```shell
+./e2e-wi-tests.sh
```
Comment (chizhg) [inline test/e2e/README.md:140]: Better to keep them consistent.
```suggestion
./test//e2e-wi-tests.sh
```

File: test/e2e/README.md [hunk 8]
```diff
@@ -42,35 +67,78 @@ go test --tags=e2e ./test/e2e/... --count=3
...
+```shell
+E2E_PROJECT_ID=<project name> go test --tags=e2e \
+  -workloadIndentity=true \
+  -pubsubServiceAccount=cre-pubsub@$PROJECT_ID.iam.gserviceaccount.com \
+  ./test/e2e/... -run NameOfTest
+```
+
+For example, to run TestPullSubscription:
+
+```shell \
```
Comment (chizhg) [inline test/e2e/README.md:109]:
```suggestion
```shell
```

File: test/e2e/README.md [hunk 9]
```diff
@@ -42,35 +67,78 @@ go test --tags=e2e ./test/e2e/... --count=3
...
-go test --tags=e2e ./test/e2e/... -run NameOfTest
+E2E_PROJECT_ID=<project name> \
+  go test --tags=e2e ./test/e2e/... -run NameOfTest
 ```
 
 For example, to run TestPullSubscription:
 
-```shell
-GOOGLE_APPLICATION_CREDENTIALS=<path to json creds file> \
+```shell \
```
Comment (chizhg) [inline test/e2e/README.md:76]:
```suggestion
```shell
```

File: test/e2e/README.md [hunk 10]
```diff
@@ -24,11 +24,36 @@ knative-gcp should be added under [knative-gcp e2e test lib](lib).
...
+1. [GCP Broker Deployment](../../docs/install/install-gcp-broker.md#deployment) and [GCP Broker Authentication Setup](../../docs/install/install-gcp-broker.md#authentication-setup-for-gcp-broker).
+1. [Broker with Pub/Sub Channel](../../docs/install/install-broker-with-pubsub-channel.md) installed.
+1. [CloudStorageSource Prerequisites](../../docs/examples/cloudstoragesource/README.md#prerequisites).
+   Note that you only need to: 
+   1. Create with an App Engine application in your project.
+1. [CloudSchedulerSource Prerequisites](../../docs/examples/cloudschedulersource/README.md#prerequisites).
+   Note that you only need to:
+   1. Enable the Cloud Storage API on your project.
+   1. Give Google Cloud Storage permissions to publish to GCP Pub/Sub.
+1. A docker repo containing [the test images](#test-images). Remember to specify the build tag `e2e`.
```
Comment (chizhg) [inline test/e2e/README.md:42]: The build tag `e2e` is auto-added by `upload-test-images.sh` so no need to mention it here.

File: test/e2e/README.md (hunk 6 above)
Comment (capri-xiyue) [reply test/e2e/README.md:172]: Does this work in Mac? When I tried to run it, it failed with 
`sed: 1: "vendor/knative.dev/even ...": invalid command code v`

File: test/e2e/README.md [hunk 11]
```diff
@@ -42,35 +67,78 @@ go test --tags=e2e ./test/e2e/... --count=3
...
 
-Note that if you plan on running metrics-related E2E tests using the StackDriver
-backend, you need to give your
-[Service Account](../../docs/install/pubsub-service-account.md) the
-`Monitoring Editor` role on your Google Cloud project:
+### Running E2E tests with authentication mechanism using Workload Identity
+```shell
+go test --tags=e2e 
+  -workloadIndentity=true \
+  -pubsubServiceAccount=cre-pubsub@$PROJECT_ID.iam.gserviceaccount.com \
```
Comment (grac3gao) [inline test/e2e/README.md:85]: maybe pointed out the `-pubsubServiceAccount=cre-pubsub@$PROJECT_ID.iam.gserviceaccount.com` is the GSA used for the resources

File: test/e2e/README.md (hunk 10 above)
Comment (capri-xiyue) [reply test/e2e/README.md:42]: It looks like it add tag `latest` insetead of `e2e` https://github.com/knative/eventing/blob/master/test/upload-test-images.sh

**Pull request merged**